src/hooks/useIsValidExecution.ts

```diff
@@ -0,0 +1,66 @@
+import { getSpecificGnosisSafeContractInstance } from '@/services/contracts/safeContracts'
+import type { SafeTransaction } from '@gnosis.pm/safe-core-sdk-types'
+import type { BigNumber } from 'ethers'
+import type { EthersError } from '@/utils/ethers-utils'
+
+import useAsync from './useAsync'
+import useSafeInfo from './useSafeInfo'
+import useWallet from './wallets/useWallet'
+import { encodeSignatures } from '@/services/tx/encodeSignatures'
+import ContractErrorCodes from '@/services/contracts/ContractErrorCodes'
+
+const isContractError = <T extends EthersError>(error: T): error is T & { reason: keyof typeof ContractErrorCodes } => {
+  return Object.keys(ContractErrorCodes).includes(error.reason)
+}
+
+const useIsValidExecution = (
+  safeTx?: SafeTransaction,
+  gasLimit?: BigNumber,
+): {
+  isValidExecution?: boolean
+  executionValidationError?: Error
+  isValidExecutionLoading: boolean
+} => {
+  const wallet = useWallet()
+  const { safe } = useSafeInfo()
+
+  const [isValidExecution, executionValidationError, isValidExecutionLoading] = useAsync(async () => {
+    if (!safeTx || !wallet?.address || !gasLimit) {
+      return
+    }
+
+    const { contract } = getSpecificGnosisSafeContractInstance(safe)
+
+    console.log('======================================== contract', contract)
+
+    try {
+      const test = await contract.callStatic.execTransaction(
```

Comment:
Leftover variable name from testing. I'll revert it when I push fixes for the test.